src/mono/wasm/runtime/library_mono.js

```diff
@@ -1507,6 +1508,12 @@ var MonoSupportLib = {
 			Module.ccall ('mono_wasm_load_profiler_coverage', null, ['string'], [arg]);
 		},
 
+		mono_wasm_register_custom_marshaler: function (fullName, marshalerFullName) {
```

Comment:
Is the intention that the embedder can interpose an arbitrary marshaller on any type in the application?

I'm wondering if it's better to use a custom attribute on the type to specify it's marshaller:

```csharp
[JavaScriptMarshaller(typeof(MyCustomClassMarshaller))]
public class CustomClass { ... }

public class MyCustomClassMarshaler {
  public static CustomClass FromJavaScript (double d) { ... }

  public static double ToJavaScript (CustomClass ct) { ... }
}
```

Reply:
Discussed on Discord.  One of the benefits of the current approach is that it is more straightforwardly usable with AOT without a lot of additional infrastructure.  And there is opportunity to go to a more declarative approach in the future.